images/openshift-ci-azure/Dockerfile

```diff
@@ -0,0 +1,28 @@
+# This Dockerfile is used by openshift CI
+
+FROM registry.ci.openshift.org/openshift/release:golang-1.16 AS builder
+WORKDIR /go/src/github.com/code-ready/crc
+COPY . .
+RUN make release
+
+FROM quay.io/centos/centos:stream8
+COPY --from=builder /go/src/github.com/code-ready/crc /opt/crc
+COPY --from=builder /go/src/github.com/code-ready/crc/out/windows-amd64/crc.exe /bin/crc.exe
+COPY --from=builder /go/src/github.com/code-ready/crc/images/openshift-ci/mock-nss.sh /bin/mock-nss.sh
+COPY --from=builder /go/src/github.com/code-ready/crc/images/openshift-ci-azure/azure-cli.repo /etc/yum.repos.d/azure-cli.repo
+
+
+RUN yum update -y && \
+    yum install --setopt=tsflags=nodocs -y \
+    nss_wrapper \
+    sshpass \
+    unzip \
+    azure-cli \
+    jq \
+    openssh-clients && \
```

Comment:
Not clear why unzip/jq are needed here, is it possible to add a link to what will be using this image to get the full picture?

Reply:
```
AZURE_AUTH_TENANT_ID=$(cat $AZURE_AUTH_LOCATION | jq -r .tenantId)
AZURE_SUBSCRIPTION_ID=$(cat $AZURE_AUTH_LOCATION | jq -r .subscriptionId)
```

But this is not part of this PR, again, very hard to review this image generation without having any clue how it's going to be used.

Reply:
@cfergeau yes it is not part of this PR because this image is going to used in the openshift-CI side so once this is merge then only I can create a PR on openshift/release side which will have those steps.

Reply:
I don't know what I'm expected to do during this review then. "Yes this is a Dockerfile which will create an image with the azure client, a crc binary, crc sources, and a few random binaries.", which is not that useful.

Reply:
By the way, maybe it would be simpler to have a single image for both gcp and azure?

Reply:
> By the way, maybe it would be simpler to have a single image for both gcp and azure?

@cfergeau yes I did think so but then we are increasing image size unnecessary to have additional packages, that's why I have it separate.

> "Yes this is a Dockerfile which will create an image with the azure client, a crc binary, crc sources, and a few random binaries.", which is not that useful.

To be honest that is only I want as review because use for this image is not in the repo so I can't provide much info.

Reply:
> then we are increasing image size unnecessary

What are the exact requirements we have on image size? If it is so important, why are we using centos/stream8 when ubi8 is about twice smaller? Why isn't there an additional base image with all the common packages between gcp and azure, and then azure and gcp reuse this base image, this would also save some space on the image repository, and on systems installing both.

Reply:
> What are the exact requirements we have on image size? I

@cfergeau I don't know if it is mentioned somewhere, I was just going with container best practice to have only stuff in single image which is required. But I think if we use same then it will also remove our maintenance burden so made the change.